src/azul/plugins/repository/tdr_anvil/__init__.py

```diff
@@ -328,12 +328,13 @@ def _primary_bundle(self, bundle_fqid: TDRAnvilBundleFQID) -> TDRAnvilBundle:
         for entity_type, typed_keys in sorted(keys_by_type.items()):
             pk_column = entity_type + '_id'
             rows = self._retrieve_entities(source.spec, entity_type, typed_keys)
-            if entity_type == 'donors':
+            if entity_type == 'donor':
                 # We expect that the foreign key `part_of_dataset_id` is
                 # redundant for biosamples and donors. To simplify our queries,
                 # we do not follow the latter during the graph traversal.
-                # Here, we validate our expectation.
-                dataset_id: Key = one(keys_by_type['datasets'])
+                # Here, we validate our expectation. Note that the key is an
+                # array for biosamples, but not for donors.
+                dataset_id: Key = one(keys_by_type['dataset'])
                 for row in rows:
                     require(row.pop('part_of_dataset_id') == dataset_id)
```

Comment:
The commit title for this hunk surprises me. The body of the `if` has a side effect that is visible beyond the `if` so I wonder why this described as just a broken requirement.

Reply:
I had mistakenly assumed that the extraneous presence of the `part_of_dataset_id` field would not affect the behavior of the indexer in any way, but now I realize that this field will now disappear from the contents of donors in the `replica` index. Would that mean that this PR now requires a reindex?

Reply:
Or do we need a separate ticket/PR to cover this bug?

Reply:
Let's factor this into a separate ticket, if possible.

Reply:
https://github.com/DataBiosphere/azul/issues/5898